ctera/src/main/java/ch/cyberduck/core/ctera/CteraAclPermissionFeature.java

```diff
@@ -0,0 +1,214 @@
+     * is not granted the "Bypass traverse checking user" right in the Group Policy snap-in.
+     * This permission does not automatically allow running program files.
+     * Directories only.
+     */
+    public static final Acl.Role TRAVERSEPERMISSION = new Acl.Role("traversepermission");
+
+    /**
+     * Create Files: Allows or denies creating files within the folder.
+     * Directories only.
+     */
+    public static final Acl.Role CREATEFILEPERMISSION = new Acl.Role("Createfilepermission");
+
+    /**
+     * Create Folders: Allows or denies creating subfolders within the folder.
+     * Directories only.
+     */
+    public static final Acl.Role CREATEDIRECTORIESPERMISSION = new Acl.Role("CreateDirectoriespermission");
+
+    static final List<Acl.Role> allCteraCustomACLRoles = Collections.unmodifiableList(Arrays.asList(
+            READPERMISSION, WRITEPERMISSION, EXECUTEPERMISSION, DELETEPERMISSION, TRAVERSEPERMISSION, CREATEFILEPERMISSION, CREATEDIRECTORIESPERMISSION
+    ));
+
+    final static List<QName> allCteraCustomACLQn = Collections.unmodifiableList(allCteraCustomACLRoles.stream().map(CteraAclPermissionFeature::toQn).collect(Collectors.toList()));
+
+    public static QName toQn(final Acl.Role role) {
+        return new QName(CTERA_NAMESPACE_URI, role.getName(), CTERA_NAMESPACE_PREFIX);
+    }
+
+    public static String toProp(final QName qn) {
+        return qn.getLocalPart();
+    }
+
+    public static Acl.Role toRole(final QName qn) {
+        return new Acl.Role(toProp(qn));
+    }
+
+    public static Acl customPropsToAcl(final Map<String, String> customProps) {
+        if(customProps.isEmpty()) {
+            return Acl.EMPTY;
+        }
+        final Acl acl = new Acl();
+        acl.addAll(new Acl.CanonicalUser());
+        for(QName qn : allCteraCustomACLQn) {
+            if(customProps.containsKey(toProp(qn))) {
+                final String val = customProps.get(toProp(qn));
+                if(Boolean.parseBoolean(val)) {
+                    acl.addAll(new Acl.CanonicalUser(), toRole(qn));
+                }
+            }
+        }
+        return acl;
+    }
+
+    public static Permission aclToPermission(final Acl acl) {
+        if(Acl.EMPTY.equals(acl)) {
+            return new Permission(700);
+        }
+        Permission.Action action = Permission.Action.none;
+        if(acl.get(new Acl.CanonicalUser()).contains(READPERMISSION)) {
+            action = action.or(Permission.Action.read);
+        }
+        if(acl.get(new Acl.CanonicalUser()).contains(WRITEPERMISSION)) {
+            action = action.or(Permission.Action.write);
+        }
+        if(acl.get(new Acl.CanonicalUser()).contains(EXECUTEPERMISSION) || acl.get(new Acl.CanonicalUser()).contains(TRAVERSEPERMISSION)) {
+            action = action.or(Permission.Action.execute);
+        }
+        // TODO CTERA-136 only user - what about group/others?
```

Comment:
@dkocher can you ocmment - is the implementation correct?

Reply:
lgtm